Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
+
+    Check.typeOf = {};
+    Check.numeric = {};
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return defaultValue(name, 'a value') + ' is required but is undefined.';
```

Comment:
Also, are we sure `name` shouldn't be required?  We only have defaults when "sensible" defaults exist and I think it is very hard to have a sensible one for a parameter.